test/party-external-rating/get-party-external-ratings.api-test.ts

```diff
@@ -0,0 +1,213 @@
+    requestToGetExternalRatingsForParty().reply(400, 'An error message from ACBS.');
+
+    const { status, body } = await api.get(getPartyExternalRatingsUrl);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
+
+  it('returns a 500 response if ACBS if ACBS returns a status code that is NOT 200 or 400', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetExternalRatingsForParty().reply(401);
+
+    const { status, body } = await api.get(getPartyExternalRatingsUrl);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
+
+  it('returns a 500 response if creating a session with the IdP times out', async () => {
+    requestToCreateASessionWithTheIdp()
+      .delay(ENVIRONMENT_VARIABLES.ACBS_TIMEOUT + 500)
+      .reply(201, '', { 'set-cookie': 'JSESSIONID=1' });
+    givenGettingAnIdTokenFromTheIdpSucceeds();
+    requestToGetExternalRatingsForParty().reply(200, externalRatingsInAcbs);
+
+    const { status, body } = await api.get(getPartyExternalRatingsUrl);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
+
+  it('returns a 500 response if getting an id token from the IdP times out', async () => {
+    givenCreatingASessionWithTheIdpSucceeds();
+    requestToGetAnIdTokenFromTheIdp()
+      .delay(ENVIRONMENT_VARIABLES.ACBS_TIMEOUT + 500)
+      .reply(200, { id_token: idToken });
+    requestToGetExternalRatingsForParty().reply(200, externalRatingsInAcbs);
+
+    const { status, body } = await api.get(getPartyExternalRatingsUrl);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
+
+  it('returns a 500 response if getting the external ratings from ACBS times out', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetExternalRatingsForParty()
+      .delay(ENVIRONMENT_VARIABLES.ACBS_TIMEOUT + 500)
+      .reply(200, externalRatingsInAcbs);
+
+    const { status, body } = await api.get(getPartyExternalRatingsUrl);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
+
+  const requestToCreateASessionWithTheIdp = (): nock.Interceptor =>
+    nock(ENVIRONMENT_VARIABLES.ACBS_AUTHENTICATION_BASE_URL)
+      .post('/sessions', {
+        loginName: ENVIRONMENT_VARIABLES.ACBS_AUTHENTICATION_LOGIN_NAME,
+        password: ENVIRONMENT_VARIABLES.ACBS_AUTHENTICATION_PASSWORD,
+      })
+      .matchHeader('content-type', 'application/json')
+      .matchHeader(ENVIRONMENT_VARIABLES.ACBS_API_KEY_HEADER_NAME, ENVIRONMENT_VARIABLES.ACBS_API_KEY);
+
+  const givenCreatingASessionWithTheIdpSucceeds = (): void => {
```

Comment:
We'll want to reuse these across tests, won't we? Might it be better to move them to a 'helpers' folder or something similar?

Reply:
@catenoid I've done this [here](https://github.com/lmarrai-sw/tfs-api/pull/2) now :)